### PR TITLE
Glob_Patterns.rst: Remove redundant examples

### PR DESCRIPTION
**Pull request by aptrishu**
It removes some unnecessary examples.

Closes https://github.com/coala/documentation/issues/70

### CODE REVIEW

**Comment by Nosferatul** [comment]
Hi! `Fixes` -> `Closes`. We use `Fixes` for bugs! The commit looks good to me!

**Comment by aptrishu** [comment]
I have updated the commit. @Nosferatul

**Comment by Nosferatul** [comment]
The commit message looks good to me 👍

File: Users/Glob_Patterns.rst [hunk 1]
```diff
@@ -123,14 +121,10 @@ brackets. Parentheses that have no match are ignored as well as
     False
```
Comment (Nosferatul) [inline Users/Glob_Patterns.rst]: We could delete this example too. I don't find the purpose of it. This is my opinion. Don't do it until someone else says that it's ok.

Comment (aptrishu) [inline Users/Glob_Patterns.rst]: It justifies the statement that [!seq] here seq cannot be empty . If the seq is empty than no matter what pattern are you matching it with,  it'll always return false. unless the string and pattern are exactly same. i.e;
if I do
fnmatch("a[!]a","a[!]a")
It will return true. :)

**Comment by aptrishu** [comment]
Please review now. @Adrianzatreanu

**Comment by sils** [comment]
CC @fneu can you take a look? This looks still like way too many examples to me. I'd remove them all and think from scratch which examples are actually useful and add them with explanations.

**Comment by aptrishu** [comment]
Should I explain every example what it stands for ? I can do that because I have kept only those examples which are necessary. @Nosferatul @sils @Adrianzatreanu

**Comment by sils** [comment]
How did you define "necessary"?

**Comment by aptrishu** [comment]
Every example that is there is only because it has a concept associated with it which can't be understood unless we show them that example. Like for example  

[!seq] here seq cannot be empty . If the seq is empty than no matter what pattern are you matching it with, it'll always return false. unless the string and pattern are exactly same. i.e;
if I do
fnmatch("a[!]a","a[!]a")
It will return true.  
fnmatch("aa","a[!]a")
returns false;
So, here I could only show the second example for the meaning of [!seq] but they need to know that if the seq and pattern are exactly the same then it'll always return true.

**Comment by sils** [comment]
IMO if every example has a meaning *as documentation* we should have some text explaining them

**Comment by sils** [comment]
CC @fneu

**Comment by Nosferatul** [comment]
Yes, we should explain some examples so that the user would understand faster.

File: Users/Glob_Patterns.rst [hunk 2]
```diff
@@ -180,8 +174,6 @@ brackets. Parentheses that have no match are ignored as well as
     True
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: I'd say we can remove L177 too (`fnmatch("abc", "a*c")`) since that is anyway covered with `abbc`.

**Comment by aptrishu** [comment]
Can you explain a little ? @Nosferatul  @sils

File: Users/Glob_Patterns.rst (hunk 2 above)
Comment (aptrishu) [inline Users/Glob_Patterns.rst]: Done @adtac . I removed that line.

**Comment by sims1253** [comment]
@aptrishu Can you in addition to the removal also make sure the whole thing is good documentation and explains itself and not just a test?

**Comment by aptrishu** [comment]
I didn't get exactly what you want. Can you elaborate a little ?
@sims1253

**Comment by sims1253** [comment]
Well it only shows a list of commands but is lacking explanatory text

**Comment by AbdealiJK** [comment]
What they want is to make it more like a story. Right now it's an info dump

So, essentially they want it more like - take a use case. Where you define a project structure and tell the user how to write globs to select certain things which a normal project would do. While doing that, explain all the stuff currently there.

**Comment by Nosferatul** [comment]
I like the idea of `story`. It should be more user friendly 👍

**Comment by aptrishu** [comment]
These are simple regex. Explaining them would make it more verbose. @Nosferatul  @AbdealiJK

**Comment by Nosferatul** [comment]
But maybe it will make more sens for the user, that's my opinion. You could try to take a structure and based on it, explain to the user how to write globs as @AbdealiJK suggested. This will show to the user the utility. If we don't like how it fits we should stick to these examples,  I believe. 👍

File: Users/Glob_Patterns.rst [hunk 3]
```diff
@@ -192,13 +182,41 @@ brackets. Parentheses that have no match are ignored as well as
...
-    >>> fnmatch("a?c", "a**c")
-    True
-    >>> fnmatch("ac", "a**c")
-    True
+
+Using Glob Patterns
+-------------------
+
+Suppose, You want `SpaceConsistencyBear` to perform an analysis on a file
+`first.c`. You should use the command:
```
Comment (Nosferatul) [inline Users/Glob_Patterns.rst]: Maybe bolding the file name should be better.

File: Users/Glob_Patterns.rst [hunk 4]
```diff
@@ -192,13 +182,41 @@ brackets. Parentheses that have no match are ignored as well as
...
+Going further, if you want all `.c` as well as `.java` files to be analysed:
+::
+
+    coala --files='src/\*.(java|c)' --bears=SpaceConsistencyBear --save
+
+If you want your `files` argument to match all directories and subdirectories,
+you can use `**` Glob Pattern for that.
+::
+
+    files = **/*.c
```
Comment (Nosferatul) [inline Users/Glob_Patterns.rst]: Maybe we should have 1-2 more examples? I'm not sure but I like it 👍

Comment (aptrishu) [inline Users/Glob_Patterns.rst]: Actually, There are plenty of them in other pages of doc. I think user would come across them and then realize how important glob patterns are. So, I left some part for the users to explore.

**Comment by Nosferatul** [comment]
Someone else should see this also. We need more opinions.

**Comment by aptrishu** [comment]
Done @Nosferatul . :+1:

File: Users/Glob_Patterns.rst [hunk 5]
```diff
@@ -192,13 +182,42 @@ brackets. Parentheses that have no match are ignored as well as
...
-    True
+
+Using Glob Patterns
+-------------------
+
+Suppose, You want `SpaceConsistencyBear` to perform an analysis on a file
+``first.c``. You should use the command:
+::
+
+    coala --files=src/\first.c --bears=SpaceConsistencyBear --save
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: what's the slash before the `f`?

File: Users/Glob_Patterns.rst [hunk 6]
```diff
@@ -192,13 +182,42 @@ brackets. Parentheses that have no match are ignored as well as
...
     True
-    >>> fnmatch("a?c", "a**c")
-    True
-    >>> fnmatch("ac", "a**c")
-    True
+
+Using Glob Patterns
+-------------------
+
+Suppose, You want `SpaceConsistencyBear` to perform an analysis on a file
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: nitpicking: no comma after suppose and lower case `y` for `you`

File: Users/Glob_Patterns.rst [hunk 7]
```diff
@@ -192,13 +182,42 @@ brackets. Parentheses that have no match are ignored as well as
...
+
+.. note::
+
+    If you don't know the fuctions of a bear or how to perform the analysis
+    with a bear, You should go through `Tutorial
+    <http://coala.readthedocs.io/en/latest/Users/Tutorial.html#ignoring
+    -issues>`_ first.
+
+Now, if you want all the ``.c`` files in a specific directory to be analysed,
+you can take help of Glob Patterns.
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: `Glob Patterns` -> `glob patterns`

File: Users/Glob_Patterns.rst [hunk 8]
```diff
@@ -192,13 +182,42 @@ brackets. Parentheses that have no match are ignored as well as
...
+::
+
+    coala --files=src/\*.c --bears=SpaceConsistencyBear --save
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: rather quote the file glob - the backslashes are distracting and might confuse the user

Comment (adtac) [inline Users/Glob_Patterns.rst]: I mean, wrap the glob with `'` and remove the backslash

File: Users/Glob_Patterns.rst [hunk 9]
```diff
@@ -192,13 +182,42 @@ brackets. Parentheses that have no match are ignored as well as
...
+
+Here, ``*.c`` matches all ``.c`` files in the specified directory.
+Going further, if you want all ``.c`` as well as ``.java`` files to
+be analysed:
+::
+
+    coala --files='src/\*.(java|c)' --bears=SpaceConsistencyBear --save
+
+If you want your `files` argument to match all directories and subdirectories,
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: double backticks around `files`

File: Users/Glob_Patterns.rst [hunk 10]
```diff
@@ -192,13 +182,42 @@ brackets. Parentheses that have no match are ignored as well as
...
+you can use `**` Glob Pattern for that.
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: again, double backticks

File: Users/Glob_Patterns.rst (hunk 4 above)
Comment (adtac) [inline Users/Glob_Patterns.rst]: Could you add a note here saying that only `.c` files at least one subdirectory will be matched? This issue has been brought up a few times by others - the basic idea is that since we have a `/` in the glob expression, any matching file will need to be at least one subdirectory below (to have the `/`). So no `.c` files in the root directory will be matched.

This is not obvious, but kinda logical. So I think this would be good mention in the docs.

File: Users/Glob_Patterns.rst [hunk 11]
```diff
@@ -192,13 +182,48 @@ brackets. Parentheses that have no match are ignored as well as
...
     True
     >>> fnmatch("a/c", "a**c")
     True
-    >>> fnmatch("a?c", "a**c")
-    True
-    >>> fnmatch("ac", "a**c")
-    True
+
+Using Glob Patterns
+-------------------
```
Comment (sils) [inline Users/Glob_Patterns.rst]: this paragraph is nice, let's put it on top so users see it and then can see the "advanced example collection" if it doesn't suffice?

File: Users/Glob_Patterns.rst [hunk 12]
```diff
@@ -192,13 +182,48 @@ brackets. Parentheses that have no match are ignored as well as
...
+
+Suppose you want `SpaceConsistencyBear` to perform an analysis on a file
+``first.c``. You should use the command:
```
Comment (sils) [inline Users/Glob_Patterns.rst]: `s/. Y/, y/`

File: Users/Glob_Patterns.rst [hunk 13]
```diff
@@ -192,13 +182,48 @@ brackets. Parentheses that have no match are ignored as well as
...
+::
+
+    coala --files=src/first.c --bears=SpaceConsistencyBear --save
+
+.. note::
+
+    If you don't know the fuctions of a bear or how to perform the analysis
+    with a bear, You should go through `Tutorial
+    <http://coala.readthedocs.io/en/latest/Users/Tutorial.html#ignoring
+    -issues>`_ first.
```
Comment (sils) [inline Users/Glob_Patterns.rst]: why link to ignoring issues paragraph? Doesn't seem to make much sense

File: Users/Glob_Patterns.rst [hunk 14]
```diff
@@ -192,13 +182,48 @@ brackets. Parentheses that have no match are ignored as well as
...
+
+Now, if you want all the ``.c`` files in a specific directory to be analysed,
+you can take help of glob patterns.
+::
+
+    coala --files=src/'*.c' --bears=SpaceConsistencyBear --save
```
Comment (sils) [inline Users/Glob_Patterns.rst]: why are there single quotes around only the *.c and not the rest? This looks pretty weird

**Comment by sils** [comment]
way better @aptrishu

File: Users/Glob_Patterns.rst (hunk 11 above)
Comment (aptrishu) [inline Users/Glob_Patterns.rst]: Yeah that would be nice.

File: Users/Glob_Patterns.rst (hunk 12 above)
Comment (aptrishu) [inline Users/Glob_Patterns.rst]: suppose You want `SpaceConsistencyBear` to perform an analysis on a file
+``first.c``. you should use the command:

Like this ? I didn't get the small and capital letters' combination.

File: Users/Glob_Patterns.rst (hunk 11 above)
Comment (aptrishu) [inline Users/Glob_Patterns.rst]: Do you want me to put this paragraph at the very beginning of the page ?

Comment (sils) [inline Users/Glob_Patterns.rst]: yes

**Comment by aptrishu** [comment]
Did the changes. @sils @adtac

**Comment by aptrishu** [comment]
Sorry, closed the pull request by mistake. I have reopened it.

File: Users/Glob_Patterns.rst [hunk 15]
```diff
@@ -1,6 +1,47 @@
...
+
+Suppose you want `SpaceConsistencyBear` to perform an analysis on a file
+``first.c``, you should use the command:
+::
+
+    coala --files=src/first.c --bears=SpaceConsistencyBear --save
+
+.. note::
+
+    If you don't know the fuctions of a bear or how to perform the analysis
```
Comment (AbdealiJK) [inline Users/Glob_Patterns.rst]: Functions - typo.

File: Users/Glob_Patterns.rst [hunk 16]
```diff
@@ -1,6 +1,47 @@
...
+    with a bear, you should go through `Tutorial
+    <http://coala.readthedocs.io/en/latest/Users/Tutorial.html>`_ first.
+
+Now, if you want all the ``.c`` files in a specific directory to be analysed,
+you can take help of glob patterns.
+::
+
+    coala --files='src/*.c' --bears=SpaceConsistencyBear --save
+
+Here, ``*.c`` matches all ``.c`` files in the specified directory.
```
Comment (AbdealiJK) [inline Users/Glob_Patterns.rst]: Mention src instead of specified directory to be clearer

File: Users/Glob_Patterns.rst [hunk 17]
```diff
@@ -1,6 +1,47 @@
...
+
+If you want your ``files`` argument to match all directories and
+subdirectories, you can use ``**`` glob pattern for that.
+::
+
+    files = **/*.c
+
+.. note::
+
+    While using ``files = **/*.c`` , since we have used ``\`` in the glob
```
Comment (AbdealiJK) [inline Users/Glob_Patterns.rst]: Unusual space before comma

Comment (AbdealiJK) [inline Users/Glob_Patterns.rst]: I don't understand this note. Where is backslash used? Why will directories below root be matched?

File: Users/Glob_Patterns.rst [hunk 18]
```diff
@@ -1,6 +1,47 @@
 Glob - Extended unix style pathname expansion
 =============================================
 
+Using Glob Patterns
+-------------------
+
+Suppose you want `SpaceConsistencyBear` to perform an analysis on a file
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: doubleticks?

File: Users/Glob_Patterns.rst (hunk 17 above)
Comment (adtac) [inline Users/Glob_Patterns.rst]: it should be `/` I think

File: Users/Glob_Patterns.rst [hunk 19]
```diff
@@ -1,6 +1,47 @@
...
+If you want your ``files`` argument to match all directories and
+subdirectories, you can use ``**`` glob pattern for that.
+::
+
+    files = **/*.c
+
+.. note::
+
+    While using ``files = **/*.c`` , since we have used ``\`` in the glob
+    pattern so all ``.c`` files at least one subdirectory below the root
```
Comment (adtac) [inline Users/Glob_Patterns.rst]: I don't think using a `so` is valid when you begin a phrase with `since`? Not sure, others can chip in.

**Comment by aptrishu** [comment]
did the changes. @AbdealiJK  @adtac

File: Users/Glob_Patterns.rst (hunk 19 above)
Comment (aptrishu) [inline Users/Glob_Patterns.rst]: No, we shouldn't. I searched at english.stackexachange.

File: Users/Glob_Patterns.rst (hunk 1 above)
Comment (jayvdb) [inline Users/Glob_Patterns.rst]: Why isnt `[!]` an error, or a noisy warning?
It is confusing to list it here, without explanation that this is invalid syntax.

File: Users/Glob_Patterns.rst [hunk 20]
```diff
@@ -174,14 +209,10 @@ brackets. Parentheses that have no match are ignored as well as
```
Comment (arjunsinghy96) [inline Users/Glob_Patterns.rst]: @aptrishu remove the redundant '\' before special characters.

File: Users/Glob_Patterns.rst (hunk 1 above)
Comment (sils) [inline Users/Glob_Patterns.rst]: CC @fneu

Comment (aptrishu) [inline Users/Glob_Patterns.rst]: So, should I add an explanation for that in there ?

Comment (aptrishu) [inline Users/Glob_Patterns.rst]: It's not a kind of error, I think. and as far as its output is concerned it's explained in the docs.

File: Users/Glob_Patterns.rst (hunk 20 above)
Comment (arjunsinghy96) [inline Users/Glob_Patterns.rst]: line 219 it is ``\*\*`` whereas it should be ``**`` (The subheading). And may other occurrences are there. As this is not related to the issue, it is up to you if you want to do it.

Comment (aptrishu) [inline Users/Glob_Patterns.rst]: Every '*' is an special character should be preceded by '\' , unless used inside the quotes.' '

Comment (arjunsinghy96) [inline Users/Glob_Patterns.rst]: **grave accent**  `` ` `` is different from Apostrophe `` ' ``. We use grave accent to highlight text not apostrophe. And under **grave accent** `*` is not a special character. Therefore, no need of `\` .

Comment (aptrishu) [inline Users/Glob_Patterns.rst]: I think the heading there displays pattern the same way it should be used. I don't think it's about highlighting thing. (it's fine `` highlights the content, and it has no trouble being used with \)  but coming to usage of glob pattern, while using in command line we should provide \ before each *.

Comment (aptrishu) [inline Users/Glob_Patterns.rst]: But following the wildcards of os , we don't need to use \ before each *.
**/ should be fine.

Comment (arjunsinghy96) [inline Users/Glob_Patterns.rst]: `\` is not required before `*` in command line i guess

Comment (aptrishu) [inline Users/Glob_Patterns.rst]: **

**Comment by sils** [comment]
ack https://github.com/coala/documentation/pull/187/commits/fd9e6d7f082af4273dd1ec811d4c081a7ca690a7

**Comment by sils** [comment]
time to get this merged @aptrishu can you rebase and ping so we merge it?

**Comment by sils** [comment]
reack 2d070c6

**Comment by sils** [comment]
@rultor merge
